web/api/queries/query.c

```diff
@@ -666,26 +666,41 @@ RRDR *rrd2rrdr(
             info("INTERNAL CHECK: %s: requested gtime %ld secs, is greater than the desired duration %ld secs", st->id, resampling_time_requested, duration);
             #endif
 
-            group = available_points; // use all the points
+            after_requested = before_requested - resampling_time_requested;
+            duration = before_requested - after_requested;
```

Comment:
Is it better for readability than `duration = resampling_time_requested`?

Reply:
This line appears 3 times. I tried to have it exactly the same in all 3 occasions. For readability.